yt/loaders.py

```diff
@@ -346,8 +351,10 @@ def load_uniform_grid(
         grid_dimensions = domain_dimensions.reshape(nprocs, 3).astype("int32")
 
     if cell_widths is not None:
-        # make sure this is a list, or else leave it as an empty guard value
-        cell_widths = [cell_widths]
+        # cell_widths left as an empty guard value if None
+        if nprocs != 1:
+            raise NotImplementedError("nprocs must equal 1 if supplying cell_widths.")
+        cell_widths = _validate_cell_widths(cell_widths, domain_dimensions)
```

Comment:
but if nprocs > 1 and cell_widths is not None, I do want it to break. Without this, the load appears to succeed but you get an error as soon as you need to access data. I'm pretty sure the issues is that the `cell_widths` need to be decomposed in the same way as the data arrays, but haven't had a chance to look in more detail and figured raising this error immediately while loading would be clearer for a user than the subsequent error. 

Here's the full details: 

On main (or here, if you remove the `raise NotImplementedError`):

```python
import numpy as np
import yt

N = 16
data = {"density": np.random.random((N, N, N))}

cell_widths = []
for _ in range(3):
    cw = np.random.random(N)
    cw /= cw.sum()
    cell_widths.append(cw)

ds = yt.load_uniform_grid(
        data,
        data["density"].shape,
        bbox=np.array([[0.0, 1.0], [0.0, 1.0], [0.0, 1.0]]),
        cell_widths=cell_widths,
        nprocs=4,
    )
```
will successfully load, but do anything with that dataset:

```python
dens = ds.all_data()[("stream", "density")]
```
and you get
```
Traceback (most recent call last):
  File "/home/chris/miniconda3/envs/yt_dev_39/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3398, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-e6b5073a1feb>", line 1, in <cell line: 1>
    dens = ds.all_data()[("stream", "density")]
  File "/home/chris/src/yt_/yt_dev/yt/yt/data_objects/static_output.py", line 1194, in all_data
    self.index
  File "/home/chris/src/yt_/yt_dev/yt/yt/data_objects/static_output.py", line 604, in index
    self._instantiated_index = self._index_class(
  File "/home/chris/src/yt_/yt_dev/yt/yt/frontends/stream/data_structures.py", line 167, in __init__
    GridIndex.__init__(self, ds, dataset_type)
  File "/home/chris/src/yt_/yt_dev/yt/yt/geometry/geometry_handler.py", line 39, in __init__
    self._setup_geometry()
  File "/home/chris/src/yt_/yt_dev/yt/yt/geometry/grid_geometry_handler.py", line 41, in _setup_geometry
    self._parse_index()
  File "/home/chris/src/yt_/yt_dev/yt/yt/frontends/stream/data_structures.py", line 209, in _parse_index
    self.grids.append(self.grid(id, self))
  File "/home/chris/src/yt_/yt_dev/yt/yt/frontends/stream/data_structures.py", line 87, in __init__
    cell_widths = index.grid_cell_widths[id - self._id_offset]
IndexError: list index out of range
```

I can open an issue for this separately...

Reply:
oh, just in case you missed it, note that the line before `if nprocs != 1:` is ` if cell_widths is not None:`, so this only errors if `cell_widths` has been set.

Reply:
Oh I see, thanks for clarifying ! Yes I think it warrants a separate issue.

Reply:
just opened an issue and added a link to the issue in the code here for future reference.